pkgs/development/tools/misc/fujprog/default.nix

```
@@ -25,13 +26,14 @@ stdenv.mkDerivation rec {
   buildInputs = [
     libftdi1
     libusb-compat-0_1
-  ];
+  ] ++ stdenv.lib.optionals stdenv.isDarwin IOKit;
```

```suggestion
  ] ++ stdenv.lib.optionals stdenv.isDarwin [ IOKit ];
```

Otherwise fails with

```
error: value is a set while a list was expected, at /Users/daniel/git/nixpkgs/pkgs/development/tools/misc/fujprog/default.nix:26:17
```

---

Fixed. Thank you, and no worries! Yes, I saw some that used optional deps added this way within a list in the source tree and some without. Makes sense that list concatenate should require a list predicate.

---

Perhaps confusingly, there is `lib.optional`, which takes a single value and `lib.optionals` which takes a list.